core/src/main/scala/org/broadinstitute/monster/ingester/core/IngestController.scala

```diff
@@ -0,0 +1,348 @@
+package org.broadinstitute.monster.ingester.core
+
+import java.time.{Instant, OffsetDateTime}
+import java.util.UUID
+
+import cats.effect.{Async, Clock, IO}
+import cats.implicits._
+import doobie._
+import doobie.implicits._
+import doobie.util.log.LogHandler
+import doobie.util.transactor.Transactor
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import org.broadinstitute.monster.ingester.core.IngestController.JobSubmission
+import org.broadinstitute.monster.ingester.core.models.ApiError.NotFound
+import org.broadinstitute.monster.ingester.core.models.{
+  ApiStatus,
+  IngestData,
+  JobData,
+  JobSummary,
+  RequestSummary
+}
+import org.broadinstitute.monster.ingester.jade.JadeApiClient
+import org.broadinstitute.monster.ingester.jade.models.{IngestRequest, JobStatus}
+
+/**
+  * Component responsible for backing Ingester's API and providing the core logic for ingesting.
+  *
+  * This component has methods to process requests and add them to the database, expand them
+  * out into jobs, submit those jobs to the Jade API, and update the status of jobs. This component
+  * also provides methods by which a user can check on the status of a request or the status of all
+  * jobs under a request.
+  *
+  * @param dbClient Client to interact with Ingester's backing database.
+  * @param jadeClient Client to interact with the Jade Data Repository API.
+  */
+class IngestController(dbClient: Transactor[IO], jadeClient: JadeApiClient)(
+  implicit clk: Clock[IO]
+) {
+
+  import DoobieInstances._
+  import Constants._
+
+  private val logger = Slf4jLogger.getLogger[IO]
+  private implicit val logHandler: LogHandler = DbLogHandler(logger)
+
+  /**
+    * Main ingest method to run all methods to create a request entry and subsequent job entries.
+    *
+    * @param ingestData the list of paths and table names to ingest from/to.
+    * @param datasetId the dataset ID to ingest into.
+    * @return the UUID of the request created.
+    */
+  def ingest(ingestData: IngestData, datasetId: UUID): IO[UUID] = {
+    for {
+      now <- getNow
+      rId <- createRequest(datasetId, now)
+      _ <- initJobs(ingestData, rId)
```

Comment:
Ahhh, I see now. `createRequest` and `initJobs` should run within a single transaction for this to be safe. If you change both to return `ConnectionIO` then you can chain them here and call `transact` once.